src/marshmallow/validate.py

```diff
@@ -109,40 +109,54 @@ def __init__(self):
         def _regex_generator(
             self, relative: bool, absolute: bool, require_tld: bool
         ) -> typing.Pattern:
-            _absolute_part = "".join(
+            hostname_variants = [
```

Comment:
Nitpick: I suppose you could use a tuple here as well

Reply:
I think the list is slightly better because there's a conditional append. If you feel strongly that it should be a tuple, I can make that change, but I'm inclined to keep it a list since it's an implementation detail.